qualtran/bloqs/mod_arithmetic/mod_addition.py

```diff
@@ -87,7 +87,9 @@ def signature(self) -> 'Signature':
     def on_classical_vals(
         self, x: 'ClassicalValT', y: 'ClassicalValT'
     ) -> Dict[str, 'ClassicalValT']:
-        return {'x': x, 'y': (x + y) % self.mod}
+        if x < self.mod and y < self.mod:
+            y = (x + y) % self.mod
+        return {'x': x, 'y': y}
```

Comment:
should we instead raise a `ValueError` for out-of-bounds inputs? (similar to e.g. `BQUInt.assert_valid_classical_val`).

Reply:
depends on how you define the bloq ... if we want the bloq to be unitary then the classical action is correct and we don't need to raise an error 

however constructions assume valid range ... I don't feel strongly towards either ... although I'm a bit leaning towards raising an error to match the construction

Reply:
I'd prefer that the classical action raises an error. The tensors-from-classical-action function would need to handle this, of course

Reply:
done